### ISSUE TITLE
4.0 version many bugs

### ISSUE THREAD

**Issue by stepanenko3** [ebess/advanced-nova-media-library#320]
I use Nova 4.2.2

- If I add media and then open the editing form again, the media is not shown
- Bugs below

![](https://i.snipboard.io/GW86Uz.jpg)

![](https://i.snipboard.io/kmdGBg.jpg)

![](https://i.snipboard.io/g07yMB.jpg)

![](https://i.snipboard.io/mbVe6I.jpg)

**Comment by mucenica-bogdan** [comment]
Thanks for reporting, working on the issues now, will create a PR for a patch release in a couple of hours.

**Comment by mucenica-bogdan** [comment]
Hey @stepanenko3, I started a PR #321 to fix the above.

There is one thing I couldn't replicate: `If I add media and then open the editing form again, the media is not shown`. 
Can you give me a screen recording or some more details on how to replicate?

On my end if I:
- upload an image -> update [RESOURCE] -> go back to the edit form of the [RESOURCE] -> I see the image.
- upload an image -> update and continue editing [RESOURCE] -> I see newly uploaded the image.

Also, in the second to last [screenshoot](https://camo.githubusercontent.com/5ae91eb6e7f6d29b99558de6d7c8b5184fcb1fa789d448f909b1714b1a8e5d9d/68747470733a2f2f692e736e6970626f6172642e696f2f673037794d422e6a7067), the one with the statistics, I see that on your end you don't get the size of the image. There is an condition (there before the migration to nova4) to only show the size if it comes set from the server.  
Is this your reported bug, the missing size?

On my tests I do get the size filled in: 
![image](https://user-images.githubusercontent.com/846097/164315426-c97601de-02d7-4fa8-a1df-1f600d17ac31.png)

**Comment by stepanenko3** [comment]
@mucenica-bogdan,
I recorded a screen with this problem
https://we.tl/t-Zb2zF7Ldg7

**Comment by mucenica-bogdan** [comment]
@stepanenko3 - thanks for the video. still can't replicate from my end but I'll try to push it some more.

in the meantime, can you try to use the package from my fork by changing your `composer.json` file and see if it works better from there?

``` json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/mucenica-bogdan/advanced-nova-media-library.git"
        }
    ],
    "require": {
        "ebess/advanced-nova-media-library": "dev-nova-4-bugfix",
    }
}
```

run `composer update` after the `composer.json` file change change.

**Comment by stepanenko3** [comment]
@mucenica-bogdan 
The problem still remains

I also noticed a number of bugs:
- "Use Existing Media" is not made by the button
- In the selection modal of existing media images, you need to set object-fit: cover; or object-fit:contain
- No window rounding (like nova 4 design)
- Header without contrast background(for example, select field white on white header)
- Close is not made by the button
- Load Next Page is not made by the button
- The detail page does not show an image
- It does not show the added image in the form after the update

My code looks like this:

```php
Images::make('Image', 'image')
    ->conversionOnIndexView('small')
    ->conversionOnDetailView('medium')
    ->conversionOnPreview('medium')
    ->conversionOnForm('small')
    ->setFileName(function ($originalFilename, $extension, $model) {
        return md5($originalFilename) . '.' . $extension;
    })
    ->showDimensions()
    ->enableExistingMedia(),
```

**Comment by stepanenko3** [comment]
@mucenica-bogdan 
And of course, a large number of bugs in dark mode

**Comment by stepanenko3** [comment]
@mucenica-bogdan 
In the crop modal, the actions are not like buttons

**Comment by mucenica-bogdan** [comment]
Thanks for the info, I'll continue to work on this tomorrow morning.

**Comment by mucenica-bogdan** [comment]
Hey @stepanenko3, I just pushed a new patch to my fork.

I think I've fixed everything reported above.

The only thing that I couldn't replicate is that missing image after upload issue. I noticed that you are using conversions to show the preview images. Are you doing the conversion in a queue? If so, do you have your queue worker up and running to process the conversions?

**Comment by bkintanar** [comment]
@mucenica-bogdan thank you for your efforts. I'll check on your changes.

**Comment by stepanenko3** [comment]
@mucenica-bogdan, @bkintanar

I don't show images on the form page because of this error

![](https://i.snipboard.io/xkzN90.jpg)

**Comment by mucenica-bogdan** [comment]
can you share with me the image you've uploaded that gives you this error?

**Comment by stepanenko3** [comment]
fixed in #325
